pkg/recipes/engine/engine.go

```diff
@@ -219,3 +212,23 @@ func (e *engine) getDriver(ctx context.Context, recipeMetadata recipes.ResourceM
 	}
 	return definition, driver, nil
 }
+
+func (e *engine) getRecipeConfigSecrets(ctx context.Context, driver recipedriver.Driver, configuration *recipes.Configuration, definition *recipes.EnvironmentDefinition) (v20231001preview.SecretStoresClientListSecretsResponse, error) {
+	secrets := v20231001preview.SecretStoresClientListSecretsResponse{}
+	driverWithSecrets, ok := driver.(recipedriver.DriverWithSecrets)
+	if ok {
```

Comment:
Why not check for `!ok` and do the rest after that `!ok` if?

Reply:
we can do that as well. but is there is a reason why checking `!ok` better?

Reply:
1. Error handling upfront
2. Improve readability and maintainability
3. Avoid deeper nesting

Reply:
updated it